### PR TITLE
feat(huff_lexer): Syntax sugar for boolean values

### PR DESCRIPTION
**Pull request by clabby**
## Overview

Adds syntax sugar for boolean values
* Lexes true as the literal `0x01`
* Lexes false as the literal `0x00`

### CODE REVIEW

File: huffup/huffup [hunk 1]
```diff
@@ -120,8 +120,8 @@ main() {
 
     # Compute the URL of the release tarball in the Huff repository.
     RELEASE_URL="https://github.com/${HUFFUP_REPO}/releases/download/${HUFFUP_TAG}/"
-    BIN_TARBALL_URL="${RELEASE_URL}huffc${HUFFUP_VERSION}_${PLATFORM}_${ARCHITECTURE}.tar.gz"
-    MAN_TARBALL_URL="${RELEASE_URL}huffc${HUFFUP_VERSION}.tar.gz"
+    BIN_TARBALL_URL="${RELEASE_URL}huff_${HUFFUP_VERSION}_${PLATFORM}_${ARCHITECTURE}.tar.gz"
+    MAN_TARBALL_URL="${RELEASE_URL}huff_man_${HUFFUP_VERSION}.tar.gz"
```
Comment (abigger87) [inline huffup/huffup:124]: y this?

**Review by abigger87 — COMMENTED** [review]
One q

File: huffup/huffup (hunk 1 above)
Comment (clabby) [reply huffup/huffup:124]: `stage` isn't synced up with `main` rn- merged `main` into this PR

https://github.com/huff-language/huff-rs/commit/a5a95868f89bc593ed38cbbc06e8913ecdac59a5

**Review by abigger87 — APPROVED** [review]
LGTM

**Pull request merged**